src/serial.rs

```diff
@@ -704,9 +743,19 @@ macro_rules! usart {
                         },
                         Rx {
                             _usart: PhantomData,
+                            ker_ck: self.ker_ck,
                         },
                     )
                 }
+
+                #[allow(unused_variables)]
+                pub fn join(tx: Tx<$USARTX>, rx: Rx<$USARTX>) -> Self {
+                    Self {
+                        usart: unsafe { core::mem::transmute::<(), $USARTX>(()) },
```

Comment:
You can also use [mem::zeroed](https://doc.rust-lang.org/stable/std/mem/fn.zeroed.html) here, which gets the job done while being a bit easier to understand

```rust
usart: unsafe { core::mem::zeroed::<$USARTX>() },
```

Even better if you add an assert to validate the assumption that `$USARTX` is a ZST.

```rust
assert_eq!(mem::size_of::<$USARTX>(), 0);
```

Reply:
The assert isn't really necessary because of E0512:
```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> mainboardfw\src\bootloader\mod.rs:25:17
   |
25 |     let gpiog = transmute::<u32, pac::GPIOG>(0);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `u32` (32 bits)
   = note: target type: `GPIOG` (0 bits)

For more information about this error, try `rustc --explain E0512`.
```

Though it'll get optimized out so no big deal.